src/Microsoft.Health.Dicom.Blob/Features/Storage/BlobFileStore.cs

```diff
@@ -172,9 +183,43 @@ public async Task DeleteFileIfExistsAsync(long version, string partitionName, Ca
         BlockBlobClient blobClient = GetInstanceBlockBlobClient(version, partitionName);
         _logger.LogInformation("Trying to delete DICOM instance file with watermark '{Version}'.", version);
 
+        EmitTelemetry("DeleteFileIfExistsAsync", OperationType.Input);
+
         await ExecuteAsync(() => blobClient.DeleteIfExistsAsync(DeleteSnapshotsOption.IncludeSnapshots, conditions: null, cancellationToken));
     }
 
+    private static readonly Action<ILogger, string, bool, Exception> LogBlobClientOperationDelegate =
+        LoggerMessage.Define<string, bool>(
+            LogLevel.Information,
+            default,
+            "Operation '{OperationName}' processed. Using external store {IsExternalStore}.");
+
+    private static readonly Action<ILogger, string, long, Exception> LogBlobClientOperationWithStreamDelegate =
+        LoggerMessage.Define<string, long>(
+            LogLevel.Information,
+            default,
+            "Operation '{OperationName}' processed stream length '{StreamLength}'.");
+
+    private void EmitTelemetry(string operationName, OperationType operationType, long? streamLength = null)
+    {
+        _blobFileStoreMeter.BlobFileStoreOperationCount.Add(
```

Comment:
Could we instead inject these sorts of basic metrics inside of the blob client's request pipeline as a policy?

Reply:
Sure, we don't need to do it if it looks like a bunch of work. It may be something of interest in the future such that each method doesn't need to explicit write the metrics.